common/workunit/workunit.cpp

```diff
@@ -13192,6 +13192,51 @@ bool isValidMemoryValue(const char *memoryUnit)
 }
 
 #ifdef _CONTAINERIZED
+bool executeGraphOnLingeringThor(IConstWorkUnit &workunit, const char *graphName)
+{
+    // check if lingering thor instance is up.
+    Owned<IStringIterator> iter = &workunit.getDebugValues("thorinstance_*");
```

Comment:
Should this be reloading the workunit (especially the debug settings) before checking.  Otherwise I think it may still have the old values (e.g. after the graph completes), so it will always be 1 and wait for 10s before it realises it has gone.

I wonder if any data used for communicating state should *not* be in the workunit, but should be stored somewhere else.  Otherwise do you need to invalidate the whole workunit to update that state?  Or is the update magically incremental?

Reply:
>Should this be reloading the workunit (especially the debug settings) before checking.

as we were talking offline - I was wondering same.. I'll look.

>I wonder if any data used for communicating state should not be in the workunit

Somewhere else in Dali for example?
It wants to be couple with the workunit somehow..


The workunit inferface has a forceReload() method, which does invalidate the whole connection cache.
The connection interface though (IRemoteConnection) can invalidate just parts of it, so could e.g. clear the Debug/ cache and would therefore force it to reload next time they were queried.

Reply:
There is precendent for wu info being stored outside the workunit because it needs to be under a different locking regime - I think (from memory) we keep the state and/or abort requests outside